### DISCUSSION TITLE
UV Tiled Textures in Spoke

### DISCUSSION THREAD

**Discussion by Ronald-MannMade** [mozilla/hubs#2860] [General Questions & Discussion] [answered]
![checker-spoke](https://user-images.githubusercontent.com/61537020/90241742-f397e700-de2b-11ea-897a-688f2f689fde.jpg)
That is in Spoke

![checker-object](https://user-images.githubusercontent.com/61537020/90241779-06122080-de2c-11ea-9d70-6ed3996a4800.jpg)
That's the glb dragged into the room

and for good measure , a gltf viewer:
![checker-viewer](https://user-images.githubusercontent.com/61537020/90241825-19bd8700-de2c-11ea-8934-0bfc71076638.jpg)

**Comment by cvanmeurs** [comment]
I need a little more info before I can help.  Can you share a screenshot of your texture node setup?  Also, do you happen to have multiple UV channels on this mesh?

**Comment by Ronald-MannMade** [comment, marked as answer]
Certainly ,
There is only one UV Channel and you can see in the Mapping node how the
scale is set to 25
![image](https://user-images.githubusercontent.com/61537020/90571384-ed9a6100-e1b1-11ea-8a92-23cffa6f2b3f.png)

On Tue, 18 Aug 2020 at 19:09, cvanmeurs <notifications@github.com> wrote:

> I need a little more info before I can help. Can you share a screenshot of
> your texture node setup? Also, do you happen to have multiple UV channels
> on this mesh?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/hubs/discussions/2860#discussioncomment-52616>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOVPV7BLNN2D4TY657UVRXTSBKYTPANCNFSM4P7L52BQ>
> .
>


-- 
Ronald Wong Ho Hip
Co-Head of Animation & VR, Technical Director,
Visual Effects Supervisor
072 160 1307 011 259 7120
ronnie@mannmade.co.za mannmade.co.za
34 Bowling Avenue, Sandton, 2090 <https://goo.gl/maps/wciUXjco9a92>
<https://www.facebook.com/mannmadepage>
<https://twitter.com/MannMadeMediaSA>
<http://www.linkedin.com/company/mann-made-media>
<https://www.youtube.com/channel/UCGXRP--FYnxsZxvr7KoVKBw>

Confidential | Important Notice: Disclaimer (forms part of the contents of
this email) The information contained in this communication from the sender
is confidential. It is intended solely for use by the recipient and others
authorized to receive it. If you are not the recipient, you are hereby
notified that any disclosure, copying, distribution or taking action in
relation of the contents of this information is strictly prohibited and may
be unlawful. Please delete it and any attachments and notify the sender
that you have received it in error. Any views or opinions presented are
solely those of the author and do not necessarily represent those of Mann
Made Media

**Reply by cvanmeurs** [reply]
Okay, I think I know what could be the issue.  When you export your GLB file, under the "Include" section of the export options, make sure "Custom Properties" is checked.  According to Jim, that's the one that will enable you to use the scaling section of the mapping node.  Otherwise you would need to manually scale your UV's 25x larger in the UV editor.  

Let me know if that works.  :)

**Reply by Ronald-MannMade** [reply]
Hey thanks for looking into this , I tried the included custom properties flag and it doesn't seem to do anything on the tiling
[MM00942 Floor Welcome Area tiling texture.zip](https://github.com/mozilla/hubs/files/5102566/MM00942.Floor.Welcome.Area.tiling.texture.zip)
Here's the blend file for reference

**Reply by cvanmeurs** [reply]
I'm getting an "invalid zip file" error when I try to extract that in Windows 10.  =/

**Reply by Ronald-MannMade** [reply]
Oh dear, maybe this works better:
https://drive.google.com/file/d/1Si0CaKGC9qoNWQ6gAsJhMiSRvKwm83jk/view?usp=sharing

**Reply by cvanmeurs** [reply]
I've been informed that there's currently a bug in Spoke that isn't properly reading UV texture transforms.  A temporary workaround is to go into your UV space and scale the actual UV islands by whatever factor you were trying to scale them in your texture transform.  (I recommend removing the texture transform values as well, so that when the bug is fixed, you don't end up having the scaling applied twice the next time you re-import your model).

The issue is being tracked here:  https://github.com/mozilla/Spoke/issues/999

Hope that helps!
